linker.py

```diff
@@ -166,6 +171,7 @@ def get_symlink_pairs(ref, *, det_map, root_map=None):
                             det_name=det_name,
                             N=point_number * fpp + fr,
                             det_type=det_type,
+                            sample=sample,
```

Comment:
```suggestion
                            sample=doc['data'].get('target_file_name', sample_name),
```
You can do this in one line and remove the code in 140-143.

Reply:
oh, nice, I updated with this